math/wide_integer/uintwide_t.h

```diff
@@ -2310,54 +2352,58 @@
-    WIDE_INTEGER_CONSTEXPR FloatingPointType extract_builtin_floating_point_type() const
+             typename = typename std::enable_if<std::is_floating_point<FloatingPointType>::value>::type>
+    WIDE_INTEGER_CONSTEXPR auto extract_builtin_floating_point_type() const -> FloatingPointType
     {
       using local_unsigned_wide_integer_type = uintwide_t<Width2, limb_type, AllocatorType, false>;
       using local_builtin_float_type         = FloatingPointType;
 
       const bool u_is_neg = is_neg(*this);
 
-      const local_unsigned_wide_integer_type u((u_is_neg == false) ? *this : -*this);
+      const local_unsigned_wide_integer_type u((!u_is_neg) ? *this : -*this);
+
+      const auto my_msb = static_cast<size_t>(msb(u));
+      const auto ilim   = size_t
+                          (
+                             size_t(  size_t(my_msb + 1U) / size_t(std::numeric_limits<limb_type>::digits))
+                           + size_t(((size_t(my_msb + 1U) % size_t(std::numeric_limits<limb_type>::digits)) != 0U) ? size_t(1U) : size_t(0U))
+                          );
 
-      const size_t my_msb = static_cast<size_t>(msb(u));
-      const size_t ilim   = size_t
-                            (
-                               size_t(  size_t(my_msb + 1U) / size_t(std::numeric_limits<limb_type>::digits))
-                             + size_t(((size_t(my_msb + 1U) % size_t(std::numeric_limits<limb_type>::digits)) != 0U) ? size_t(1U) : size_t(0U))
-                            );
+      auto a = local_builtin_float_type(0.0F);
 
-      local_builtin_float_type a = local_builtin_float_type(0.0F);
+      constexpr long double one_ldbl(1.0L);
 
-      long double ldexp_runner = 1.0L;
+      long double ldexp_runner(one_ldbl);
 
-      for(size_t i = size_t(0U); i < ilim; ++i)
+      for(auto i = size_t(0U); i < ilim; ++i)
       {
-        long double ld      = 0.0L;
-        limb_type   lm_mask = limb_type(1ULL);
+        auto ld      = static_cast<long double>(0.0L);
+        auto lm_mask = limb_type(1ULL);
 
-        for(size_t j = size_t(0U); j < size_t(std::numeric_limits<limb_type>::digits); ++j)
+        for(auto j = size_t(0U); j < size_t(std::numeric_limits<limb_type>::digits); ++j)
         {
           if(limb_type(*(u.values.cbegin() + size_t(i)) & lm_mask) != limb_type(0U))
           {
             ld = static_cast<long double>(ld + ldexp_runner);
           }
 
+          constexpr long double two_ldbl(2.0L);
+
           lm_mask      = limb_type(lm_mask << 1U);
-          ldexp_runner = static_cast<long double>(ldexp_runner * 2.0L);
+          ldexp_runner = static_cast<long double>(ldexp_runner * two_ldbl);
         }
 
-        a += (local_builtin_float_type) ld;
+        a += static_cast<local_builtin_float_type>(ld);
       }
 
-      return local_builtin_float_type((u_is_neg == false) ? a : -a);
+      return local_builtin_float_type((!u_is_neg) ? a : -a);
     }
     #endif
 
```

Comment:
I got tired here (it's 11:40pm) and need to rest! I've left a lot of comments because I've been through all this on several code bases. However, it's a small proportion of the total changes. Overall this change looks great.